rewrite-java-test/src/test/java/org/openrewrite/java/ReplaceStringLiteralWithConstantTest.java

```diff
@@ -105,17 +124,137 @@ private String method() {
               """,
             """
               package com.abc;
-
-              import com.constant.B;
-
+                          
+              import org.openrewrite.java.ReplaceStringLiteralWithConstantTest;
+                          
               class A {
-                  String v = B.VAR;
+                  String v = ReplaceStringLiteralWithConstantTest.EXAMPLE_STRING_CONSTANT;
                   private String method() {
-                      return B.VAR;
+                      return ReplaceStringLiteralWithConstantTest.EXAMPLE_STRING_CONSTANT;
                   }
               }
               """
           )
         );
     }
+
+    @Test
+    void replaceStringLiteralWithConstantValueWhenLiteralValueIsNotConfigured() {
+        rewriteRun(
+          spec -> spec.recipe(new ReplaceStringLiteralWithConstant(EXAMPLE_STRING_FQN)),
+          java(
+            """ 
+              class Test {
+                  Object o = "Hello World!";
+              }
+              """,
+            """
+              import org.openrewrite.java.ReplaceStringLiteralWithConstantTest;
+                          
+              class Test {
+                  Object o = ReplaceStringLiteralWithConstantTest.EXAMPLE_STRING_CONSTANT;
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void replaceEmptyStringLiteralWhenLiteralValueIsConfiguredEmpty() {
+        rewriteRun(
+          spec -> spec.recipe(new ReplaceStringLiteralWithConstant("", EXAMPLE_STRING_FQN)),
+          java(
+            """
+              class Test {
+                  Object o = "";
+              }
+              """,
+            """
+              import org.openrewrite.java.ReplaceStringLiteralWithConstantTest;
+                          
+              class Test {
+                  Object o = ReplaceStringLiteralWithConstantTest.EXAMPLE_STRING_CONSTANT;
+              }
+              """
+          )
+        );
+    }
```

Comment:
Well spotted inconsistency. (The testcase actually passes).

Currently we only validate that the constant is neither `null` nor empty. We do not have any validations regarding the literal value. During validation basically only an empty string is invalid, because `null` would be finde as long the constant can be validated successfully. 
Now, this led me only inserting a check as follows :
```java
if ("".equals(this.literalValue)) {
     return result.and(invalid("literalValue", literalValue, "Unsupported literal value provided. Literal value is an empty string"));
}
```
This snippet looks distracting in my opinion. Next approach would validate literalValue as follows:
```java
if (StringUtils.isNullOrEmpty(getLiteralValue())) {
    return result.and(invalid("literalValue", literalValue, "Unsupported literal value provided. Literal value is an empty string"));
}
```
This was even worse, because when missconfigured, this call would already throw exceptions. If we catch the `IllegalArgumentException`, we would give the user wrong hints, as potentially it is not the literal value being missconfigured, but rather the constant's FQN. Additionally we would possible validate the constant's FQN twice under the hood.

Long story short, I propose that we do not synthetically forbid empty string literals. Validation wise this would mean, that any literal value is valid, because it is nullable from configuration point of view plus the constant should be accessible and it's value may not be null. If you prefer to keep the rule that empty strings are invalid, then I would use the first snippet. Depending on your thoughts, I would implement and potentially remove this test case.

Reply:
Glad to see you've given it so much thought; agree with you that it's not preferable to add special handling here. We're going to have to rely on the `Option` field returning `null` when not configured, which I'm fairly certain it does, as seen in `replaceStringLiteralWithConstantValueWhenLiteralValueIsNotConfiguredYaml()`.

Reply:
Then still fine to prevent confusion by removing this test right? As it's not a case we want to explicitly support (or prevent).

And anything else before we merge your PR? Great work on this!

Reply:
I just aligned validation of `literalValue` and the constant's value and added a check that the FQN is not blank to throw a better validation error in that case.